support/k8s/k8s-workload-registrar/mode-crd/webhook/webhook_svid_watcher.go

```diff
@@ -0,0 +1,177 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	certsFileMode      = os.FileMode(0644)
+	keyFileMode        = os.FileMode(0600)
+	certsFileName      = "tls.crt"
+	keyFileName        = "tls.key"
+	trustBundleTimeout = 3 * time.Minute
+)
+
+type SVIDWatcherConfig struct {
+	AgentSocketPath string
+	Ctx             context.Context
+	Log             logrus.FieldLogger
+	SpiffeID        *types.SPIFFEID
```

Comment:
nit: this is a fairly inconvenient type; i'd suggest using `spiffeid.ID` where possible except when interacting with the API. For example, in this watcher, having this be a spiffeid.ID would allow a simple `==` comparison when finding the candidate SVID below.